### PR TITLE
HADOOP-17138. Fix spotbugs warnings surfaced after upgrade to 4.0.6.

### PR DESCRIPTION
**Pull request by tasanuma**
…(#2155)

(cherry picked from commit 1b29c9bfeee0035dd042357038b963843169d44c)

## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HADOOP-XXXXX. Fix a typo in YYY.)
For more details, please see https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute

### CODE REVIEW

**Comment by tasanuma** [comment]
There are some spotbugs warnings in #2800.

As branch-3.2 uses spotbugs after HADOOP-16870, we need to backport HADOOP-17138 to branch-3.2.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java [hunk 1]
```diff
@@ -3716,7 +3716,7 @@ void incrUserConnections(String user) {
         if (count == null) {
           count = 1;
         } else {
-          count++;
```
Comment (goiri) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java:3719]: Dead local store?
Can someone educate me on why this is an issue?

Comment (aajisaka) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java:3719]: This change is a workaround for the bug of the javac compiler.
https://github.com/spotbugs/spotbugs/issues/571#issuecomment-368337883

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   9m 23s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-3.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   3m 58s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  23m 44s |  |  branch-3.2 passed  |
| +1 :green_heart: |  compile  |  15m 32s |  |  branch-3.2 passed  |
| +1 :green_heart: |  checkstyle  |   2m 32s |  |  branch-3.2 passed  |
| +1 :green_heart: |  mvnsite  |   8m 43s |  |  branch-3.2 passed  |
| +1 :green_heart: |  javadoc  |   5m 57s |  |  branch-3.2 passed  |
| -1 :x: |  spotbugs  |   9m 56s | [/branch-spotbugs-hadoop-yarn-project_hadoop-yarn-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-yarn-project_hadoop-yarn-warnings.html) |  hadoop-yarn-project/hadoop-yarn in branch-3.2 has 2 extant spotbugs warnings.  |
| -1 :x: |  spotbugs  |   1m 52s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-3.2 has 2 extant spotbugs warnings.  |
| -1 :x: |  spotbugs  |   2m 45s | [/branch-spotbugs-hadoop-hdfs-project_hadoop-hdfs-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-hdfs-project_hadoop-hdfs-warnings.html) |  hadoop-hdfs-project/hadoop-hdfs in branch-3.2 has 4 extant spotbugs warnings.  |
| -1 :x: |  spotbugs  |   3m 49s | [/branch-spotbugs-hadoop-mapreduce-project-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-mapreduce-project-warnings.html) |  hadoop-mapreduce-project in branch-3.2 has 2 extant spotbugs warnings.  |
| -1 :x: |  spotbugs  |   0m 45s | [/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html) |  hadoop-tools/hadoop-sls in branch-3.2 has 1 extant spotbugs warnings.  |
| -1 :x: |  spotbugs  |   0m 44s | [/branch-spotbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-timelineservice-hbase-tests-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/branch-spotbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-timelineservice-hbase-tests-warnings.html) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase-tests in branch-3.2 has 2 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  12m 41s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   9m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 53s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  14m 53s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   9m  3s |  |  the patch passed  |
| +1 :green_heart: |  xml  |   0m  3s |  |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   5m 57s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m  2s |  |  hadoop-common-project/hadoop-common generated 0 new + 0 unchanged - 2 fixed = 0 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   2m 55s |  |  hadoop-hdfs-project/hadoop-hdfs generated 0 new + 0 unchanged - 4 fixed = 0 total (was 4)  |
| +1 :green_heart: |  spotbugs  |  10m  5s |  |  hadoop-yarn-project/hadoop-yarn generated 0 new + 0 unchanged - 2 fixed = 0 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   0m 54s |  |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase-tests generated 0 new + 0 unchanged - 2 fixed = 0 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   4m  0s |  |  hadoop-mapreduce-project generated 0 new + 0 unchanged - 2 fixed = 0 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   0m 58s |  |  hadoop-tools/hadoop-sls generated 0 new + 0 unchanged - 1 fixed = 0 total (was 1)  |
| +1 :green_heart: |  shadedclient  |  13m  9s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  15m 41s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch failed.  |
| -1 :x: |  unit  | 172m 25s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt) |  hadoop-hdfs in the patch failed.  |
| -1 :x: |  unit  |  48m 49s | [/patch-unit-hadoop-yarn-project_hadoop-yarn.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt) |  hadoop-yarn in the patch failed.  |
| +1 :green_heart: |  unit  |  12m 41s |  |  hadoop-yarn-server-timelineservice-hbase-tests in the patch passed.  |
| -1 :x: |  unit  |  20m 10s | [/patch-unit-hadoop-mapreduce-project.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/patch-unit-hadoop-mapreduce-project.txt) |  hadoop-mapreduce-project in the patch failed.  |
| +1 :green_heart: |  unit  |  11m  8s |  |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  5s |  |  The patch does not generate ASF License warnings.  |
|  |   | 463m 50s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.hdfs.TestDFSInotifyEventInputStreamKerberized |
|   | hadoop.yarn.server.webproxy.amfilter.TestAmFilter |
|   | hadoop.yarn.server.webproxy.TestWebAppProxyServlet |
|   | hadoop.mapreduce.split.TestJobSplitWriterWithEC |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell xml |
| uname | Linux cdc60bc9fdaf 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.2 / 5a68e20b31a48943a35ee8a19e756f69b3a9bc45 |
| Default Java | Private Build-1.8.0_282-8u282-b08-0ubuntu1~18.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/testReport/ |
| Max. process+thread count | 2590 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs hadoop-yarn-project/hadoop-yarn hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase-tests hadoop-mapreduce-project hadoop-tools/hadoop-sls U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2806/1/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Review by aajisaka — APPROVED** [review]

**Comment by tasanuma** [comment]
Thanks for your review, @aajisaka. The failed tests succeeded in my local environment.

**Pull request merged**